python-orocos-kdl: errors configuring

I am trying to create a recipe for `python-orocos-kdl` [1].

However, I am encountering following configure error:

```
| /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/x86_64-linux/usr/bin/python-native/python: can't open file '_find_sip_py-NOTFOUND': [Errno 2] No such file or directory
| CMake Error at cmake/FindSIP.cmake:54 (MESSAGE):
|   Could not find SIP
```

Here is the initial recipe:

```
DESCRIPTION = "This package contains the python bindings PyKDL for the Kinematics and Dynamics Library (KDL), distributed by the Orocos Project."
SECTION = "devel"
LICENSE = "LGPLv2"
LIC_FILES_CHKSUM = "file://package.xml;beginline=9;endline=9;md5=d566ef916e9dedc494f5f793a6690ba5"

DEPENDS = "python-sip orocos-kdl"

require kdl.inc
```

and `kdl.inc` (centralizing common things with `orocos-kdl` recipe):

```
SRC_URI = "git://github.com/orocos/orocos_kinematics_dynamics.git"
SRCREV = "15fb082c10b58078841c14c067ac55f592733447"
PV = "1.1.102+git${SRCREV}"

S = "${WORKDIR}/git/${@d.getVar('BPN', True).replace('-', '_')}"

inherit cmake

ROS_SPN = "kdl"
```

Compiling the original recipe above, gives the following error:

```
| CMake Error at /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/x86_64-linux/usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
|   Could NOT find PythonInterp (missing: PYTHON_EXECUTABLE)
| Call Stack (most recent call first):
|   /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/x86_64-linux/usr/share/cmake-2.8/Modules/FindPackageHandleStandardArgs.cmake:315 (_FPHSA_FAILURE_MESSAGE)
|   /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/x86_64-linux/usr/share/cmake-2.8/Modules/FindPythonInterp.cmake:139 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
|   CMakeLists.txt:9 (find_package)
```

I am able to "fix" this error by adding the following line at the end in my recipe:

```
inherit pythonnative
```

With this added line, I get the following error:

```
| -- Found PythonInterp: /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/x86_64-linux/usr/bin/python-native/python (found version "2.7.3")
| -- Found PythonLibs: /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/cubieboard2/usr/lib/libpython2.7.so (found version "2.7.3")
| /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/x86_64-linux/usr/bin/python-native/python: can't open file '_find_sip_py-NOTFOUND': [Errno 2] No such file or directory
| CMake Error at cmake/FindSIP.cmake:54 (MESSAGE):
|   Could not find SIP
| Call Stack (most recent call first):
```

I am at loss solving this one. I checked `FindSIP.cmake` [2], and the problem is in this line:

```
FIND_FILE(_find_sip_py FindSIP.py PATHS ${CMAKE_MODULE_PATH})
```

Apparently the find file command does not find FindSIP.py in the specified PATHS, and therefore is unable to populate `_find_sip_py` with the right value, which should be 

```
/home/kristof/OE/oe-core/build/tmp-eglibc/work/armv7a-vfp-neon-oe-linux-gnueabi/python-orocos-kdl/1.1.102+git15fb082c10b58078841c14c067ac55f592733447-r0/git/python_orocos_kdl/cmake/FindSIP.py")
```

Interestingly, `CMAKE_MODULE_PATH` contains the right search path when printing it out just before the `FIND_FILE` line:

```
CMAKE_MODULE_PATH: /home/kristof/OE/oe-core/build/tmp-eglibc/sysroots/cubieboard2/usr/share/cmake/Modules/;/home/kristof/OE/oe-core/build/tmp-eglibc/work/armv7a-vfp-neon-oe-linux-gnueabi/python-orocos-kdl/1.1.102+git15fb082c10b58078841c14c067ac55f592733447-r0/git/python_orocos_kdl/cmake
```

Any idea?

Thanks!

Kristof

[1] https://github.com/orocos/orocos_kinematics_dynamics/tree/master/python_orocos_kdl
[2] https://raw.github.com/orocos/orocos_kinematics_dynamics/master/python_orocos_kdl/cmake/FindSIP.cmake

--- a/recipes-extended/kdl/kdl.inc
+++ b/recipes-extended/kdl/kdl.inc
@@ -1,0 +1,9 @@
+SRC_URI = "git://github.com/orocos/orocos_kinematics_dynamics.git"
+SRCREV = "15fb082c10b58078841c14c067ac55f592733447"
+PV = "1.1.102+git${SRCREV}"
+
+S = "${WORKDIR}/git/${@d.getVar('BPN', True).replace('-', '_')}"
+
+inherit cmake
+
+ROS_SPN = "kdl"

--- a/recipes-extended/kdl/orocos-kdl_1.1.102-1.bb
+++ b/recipes-extended/kdl/orocos-kdl_1.1.102-1.bb
@@ -3,13 +3,7 @@ SECTION = "devel"
 LICENSE = "LGPLv2"
 LIC_FILES_CHKSUM = "file://COPYING;md5=a8ffd58e6eb29a955738b8fcc9e9e8f2"
 
-SRC_URI = "git://github.com/orocos/orocos_kinematics_dynamics.git"
-SRCREV = "15fb082c10b58078841c14c067ac55f592733447"
-PV = "1.1.102+git${SRCREV}"
-
-S = "${WORKDIR}/git/orocos_kdl"
-
-inherit cmake
+require kdl.inc
 
 FILES_${PN}-dev += "/usr/share/orocos_kdl/package.xml /usr/share/orocos_kdl/orocos_kdl-config.cmake"
 

--- a/recipes-extended/kdl/orocos-kdl_1.1.102-1.bb
+++ b/recipes-extended/kdl/orocos-kdl_1.1.102-1.bb
@@ -3,12 +3,14 @@ SECTION = "devel"
 LICENSE = "LGPLv2"
 LIC_FILES_CHKSUM = "file://COPYING;md5=a8ffd58e6eb29a955738b8fcc9e9e8f2"
 
-require kdl.inc
+DEPENDS = "libeigen"
 
-FILES_${PN}-dev += "/usr/share/orocos_kdl/package.xml /usr/share/orocos_kdl/orocos_kdl-config.cmake"
+require kdl.inc
 
 do_install_append() {
         # remove sysroot library path from pkgconfig files
         sed -i -e 's#${STAGING_INCDIR}#${includedir}#g' \
                    ${D}${libdir}/pkgconfig/*.pc
 }
+
+FILES_${PN}-dev += "${datadir}/orocos_kdl/*"

--- a/recipes-extended/kdl/python-orocos-kdl/0001-findSIP-fix.patch
+++ b/recipes-extended/kdl/python-orocos-kdl/0001-findSIP-fix.patch
@@ -1,0 +1,34 @@
+diff --git a/cmake/FindSIP.cmake b/cmake/FindSIP.cmake
+index 53e2888..8f94cf5 100644
+--- a/cmake/FindSIP.cmake
++++ b/cmake/FindSIP.cmake
+@@ -31,17 +31,26 @@ IF(SIP_VERSION)
+   SET(SIP_FOUND TRUE)
+ ELSE(SIP_VERSION)
+ 
+-  FIND_FILE(_find_sip_py FindSIP.py PATHS ${CMAKE_MODULE_PATH})
++  FIND_FILE(_find_sip_py FindSIP.py PATHS ${CMAKE_MODULE_PATH} NO_CMAKE_FIND_ROOT_PATH)
+ 
+   EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} ${_find_sip_py} OUTPUT_VARIABLE sip_config)
+   IF(sip_config)
+     STRING(REGEX REPLACE "^sip_version:([^\n]+).*$" "\\1" SIP_VERSION ${sip_config})
+     STRING(REGEX REPLACE ".*\nsip_version_str:([^\n]+).*$" "\\1" SIP_VERSION_STR ${sip_config})
+-    STRING(REGEX REPLACE ".*\nsip_bin:([^\n]+).*$" "\\1" SIP_EXECUTABLE ${sip_config})
+     IF(NOT SIP_DEFAULT_SIP_DIR)
+         STRING(REGEX REPLACE ".*\ndefault_sip_dir:([^\n]+).*$" "\\1" SIP_DEFAULT_SIP_DIR ${sip_config})
+     ENDIF(NOT SIP_DEFAULT_SIP_DIR)
+-    STRING(REGEX REPLACE ".*\nsip_inc_dir:([^\n]+).*$" "\\1" SIP_INCLUDE_DIR ${sip_config})
++
++    IF(CMAKE_CROSSCOMPILING)
++      FIND_PROGRAM(SIP_EXECUTABLE sip)
++      STRING(REGEX REPLACE ".*\nsip_inc_dir:([^\n]+).*$" "\\1" SIP_INCLUDE_DIR ${sip_config})
++      LIST(GET CMAKE_FIND_ROOT_PATH 0 SIP_SYSROOT)
++      SET(SIP_INCLUDE_DIR "${SIP_SYSROOT}${SIP_INCLUDE_DIR}")
++    ELSE(CMAKE_CROSSCOMPILING)
++      STRING(REGEX REPLACE ".*\nsip_bin:([^\n]+).*$" "\\1" SIP_EXECUTABLE ${sip_config})
++      STRING(REGEX REPLACE ".*\nsip_inc_dir:([^\n]+).*$" "\\1" SIP_INCLUDE_DIR ${sip_config})
++    ENDIF(CMAKE_CROSSCOMPILING)
++
+     SET(SIP_FOUND TRUE)
+   ENDIF(sip_config)
+ 

--- a/recipes-extended/kdl/python-orocos-kdl_1.1.102-1.bb
+++ b/recipes-extended/kdl/python-orocos-kdl_1.1.102-1.bb
@@ -1,0 +1,18 @@
+DESCRIPTION = "This package contains the python bindings PyKDL for the Kinematics and Dynamics Library (KDL), distributed by the Orocos Project."
+SECTION = "devel"
+LICENSE = "LGPLv2"
+LIC_FILES_CHKSUM = "file://package.xml;beginline=9;endline=9;md5=46ee8693f40a89a31023e97ae17ecf19"
+
+DEPENDS = "sip-native python-sip orocos-kdl"
+
+require kdl.inc
+
+SRC_URI += "file://0001-findSIP-fix.patch"
+
+inherit pythonnative
+
+FILES_${PN} += "${libdir}/python*/dist-packages/PyKDL.so"
+FILES_${PN}-dev += "${datadir}/python_orocos_kdl/*"
+FILES_${PN}-dbg += "${libdir}/python*/dist-packages/.debug/*"
+
+RDEPENDS_{PN} = "python-sip"

